### PR TITLE
config: migrate off `config.merge()`

### PR DESCRIPTION
**Pull request by yuja**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Comment by martinvonz** [comment]
Since the 0.23 release is tomorrow, should we wait with this until after that? This seems like something that could use more than a day of testing. What do people think?

File: cli/src/config.rs [hunk 1]
```diff
@@ -693,7 +693,8 @@ mod tests {
 
     #[test]
     fn test_command_args() {
-        let config = config::Config::builder()
+        let mut config = StackedConfig::empty();
+        let data = config::Config::builder()
```
[the remark this answers is not in the document]
Comment (yuja) [reply cli/src/config.rs:697]: Ah, yes. I somehow made it use `.insert_value()` API in later patch, but `parse()` is probably better here.

**Review by yuja — COMMENTED** [review]
> Since the 0.23 release is tomorrow, should we wait with this until after that? This seems like something that could use more than a day of testing.

I'm fine in either way. A bigger change `settings.get() -> stacked_config.get()` is already in, so this PR has relatively small impact. However, it doesn't provide any UX improvement at this point.

**Comment by martinvonz** [comment]
> I'm fine in either way.

I was thinking that this PR was migrating off the `config` crate when I asked (before I reviewed it), and then I didn't think to withdraw my question. I'm fine with merging this PR before the release.

**Review by martinvonz — APPROVED** [review]

**Comment by yuja** [comment]
> I was thinking that this PR was migrating off the config crate when I asked (before I reviewed it)

That would definitely require testing. It'll also include minor behavior changes.

**Review by martinvonz — APPROVED** [review]

**Pull request merged**